File: user-service/src/main/java/com/github/manosbatsis/services/user/repository/UserRepository.java
package com.github.manosbatsis.services.user.repository;

import com.github.manosbatsis.services.user.model.User;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {

    Optional<User> findUserByEmail(String email);
}

File: user-service/src/main/kotlin/com/github/manosbatsis/services/user/repository/UserRepository.kt
package com.github.manosbatsis.services.user.repository

import com.github.manosbatsis.services.user.model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Long> {
    fun findUserByEmail(email: String): User?
}
